arch/risc-v/src/common/riscv_exception_common.S

```diff
@@ -178,16 +181,8 @@ exception_common:
   .type   g_intstackalloc, object
   .type   g_intstacktop, object
 g_intstackalloc:
-#ifndef CONFIG_SMP
-  .skip  ((CONFIG_ARCH_INTERRUPTSTACK + 8) & ~15)
-#else
-  .skip  (((CONFIG_ARCH_INTERRUPTSTACK * CONFIG_SMP_NCPUS) + 8) & ~15)
```

Comment:
why not use CONFIG_SMP_NCPUS?

Reply:
The existence of this macro depends on CONFIG_SMP. I need multiple IRQ stacks in MPFS bootloader build, where it is not using SMP. It only boots nuttx on one core, but the startup logic still handles SW interrupts for other harts (the other harts end up in WFI, and continue booting only after when they receive the interrupt from the "bootloader" hart).

So, I figured that the best way is to just have an own config value for this, which defaults to SMP_NCPUS when SMP is selected. Otherwise board specific Kconfig files can override the default if needed.

Reply:
I am actually in my setup booting two NuttX instances on the  same SoC, one on e51 hart, which initializes the HW and triggers the other OSs, and another one starting on one of the u54 harts. None of those is running in SMP mode, but the one running on e51 needs the multiple irq stacks in the process of setting up the others.

Reply:
How about we move SMP_NCPUS out of if SMP?
```
config SMP_NCPUS
	int "Number of CPUs"
	default 4 if SMP
        default 1 if !SMP
	range 1 32
	---help---
		This value identifies the number of CPUs supported by the processor
		that will be used for SMP.
		If CONFIG_DEBUG_FEATURES is enabled, then the value one is permitted
		for CONFIG_SMP_NCPUS.  This is not normally a valid setting for an
		SMP configuration.  However, running the SMP logic in a single CPU
		configuration is useful during certain testing.
```
So, you can config SMP_NCPUS to other value without enabling SMP.

Reply:
I will give it a try. I was a bit scared of that, since CONFIG_SMP_NCPUS can be found in so many places and in all architectures.

Reply:
I can make it work; it is just a bit messy.  Will need something like:

#ifndef CONFIG_SMP
#  undef CONFIG_SMP_NCPUS
#  define CONFIG_SMP_NCPUS       1
#endif

At least in sched/init/nx_start.c, since I don't want it to affect there. I just need the irq stacks, not SMP CPUs

Reply:
no, it gets too scary. Maybe I will add some ARCH specific flag for this? CONFIG_ARCH_N_IRQ_STACKS, which can be defined only for mpfs for now?

Reply:
That's fine, but I still need 5 interrupt stacks even if SMP_NCPUS == 1 . This is what I am trying to accomplish here.

Reply:
But, you can set SMP_NCPUS to 5 without enable SMP now.

Reply:
No, sorry. It would allocate all sched structs 5 times, go through all the loops e.g. in herehttps://github.com/apache/incubator-nuttx/blob/f2ee1c5b35a0fdd4c06b3daf4435194cef2757de/sched/init/nx_start.c#L378

and  in many other places.

I only need 5 irq stacks for the mpfs_head.S use, I don't need 5 SMP CPUs.

It would also be extremely confusing to set SMP_NCPUS to > 1, when not having SMP, imho

Should I try not to use to common irqstacks, but define my own irqstacks somewhere else then? This is also confusing since then the exception_common would allocate one stack which wouldn't be used at all...

Reply:
Another solution that I can make is not to use the riscv_exception.S_common, but keep this file branched for mpfs: then we have control over the startup logic the way we need.

Things already broke for us badly previously in the commit 
 e47a915f4c129240bb5ed6b8e6e9bcec6c2342b2 which forces everyone to place the whole __start into the beginning of the image (we previously only had a single branch to actual starup there). I don't want to complicate things more now, but rather then just have own code for mpfs, if it is not possible to harmonize with the rest of risc-v targets.

Reply:
Yes, you are right. It's hard to mix your case with SMP. I will refine my patch to fix CONFIG_SMP_NCPUS to 1 in no SMP case, instead of undefined macro.